src/main/java/com/eprosima/fastdds/idl/templates/TypeObjectTestingTestSource.stg

```diff
@@ -0,0 +1,1101 @@
+    $elseif(parameter.typecode.isUint16Type)$
+    ASSERT_EQ($parameter.value$, $annotation.name$_$relation_name$_$parameter.name$_default_value.uint16_value());
+    $elseif(parameter.typecode.isInt32Type)$
+    ASSERT_EQ($parameter.value$, $annotation.name$_$relation_name$_$parameter.name$_default_value.int32_value());
+    $elseif(parameter.typecode.isUint32Type)$
+    ASSERT_EQ($parameter.value$, $annotation.name$_$relation_name$_$parameter.name$_default_value.uint32_value());
+    $elseif(parameter.typecode.isInt64Type)$
+    ASSERT_EQ($parameter.value$, $annotation.name$_$relation_name$_$parameter.name$_default_value.int64_value());
+    $elseif(parameter.typecode.isUint64Type)$
+    ASSERT_EQ($parameter.value$, $annotation.name$_$relation_name$_$parameter.name$_default_value.uint64_value());
+    $elseif(parameter.typecode.isFloat32Type)$
+    ASSERT_EQ($parameter.value$, $annotation.name$_$relation_name$_$parameter.name$_default_value.float32_value());
+    $elseif(parameter.typecode.isFloat64Type)$
+    ASSERT_EQ($parameter.value$, $annotation.name$_$relation_name$_$parameter.name$_default_value.float64_value());
+    $elseif(parameter.typecode.isFloat128Type)$
+    ASSERT_EQ($parameter.value$, $annotation.name$_$relation_name$_$parameter.name$_default_value.float128_value());
+    $elseif(parameter.typecode.isCharType)$
+    ASSERT_EQ($parameter.value$, $annotation.name$_$relation_name$_$parameter.name$_default_value.char_value());
+    $elseif(parameter.typecode.isWCharType)$
+    ASSERT_EQ($parameter.value$, $annotation.name$_$relation_name$_$parameter.name$_default_value.wchar_value());
+    $endif$
+    $elseif(parameter.typecode.isEnumType)$
+    ASSERT_EQ($parameter.value$, $annotation.name$_$relation_name$_$parameter.name$_default_value.enumerated_value());
+    $elseif(parameter.typecode.isStringType)$
+    ASSERT_EQ("$parameter.value$", $annotation.name$_$relation_name$_$parameter.name$_default_value.string8_value());
+    $elseif(parameter.typecode.isWStringType)$
+    ASSERT_EQ("$parameter.value$", $annotation.name$_$relation_name$_$parameter.name$_default_value.string16_value());
+    $endif$
+    memberId++;
+>>
+
+check_const_type(ctx, const, relation, relation_name, relation_capname) ::= <<
+    //=====check_const_type - $relation_name$ - $const.name$=====
+    $get_type_identifier(ctx=ctx, type=const, relation=relation, relation_name=relation_name, var_name=["type_identifier"])$
+    ASSERT_EQ($const.name$_$relation_name$_type_identifier->_d(), $const.name$_$relation_name$_type_identifier->_d());
+>>
+
+check_array_type(ctx, array, relation, relation_name, relation_capname) ::= <<
+    //=====check_array_type - $relation_name$ - $array.name$=====
+    $get_type_identifier(ctx=ctx, type=array, relation=relation, relation_name=relation_name, var_name=["type_identifier"])$
+    ASSERT_EQ(eprosima::fastrtps::types::EK_$if(relation)$COMPLETE$else$MINIMAL$endif$, $array.name$_$relation_name$_type_identifier->_d());
```

Comment:
This is not correct. Array TypeIdentifier has `TI_PLAIN_ARRAY_SMALL` or `TI_PLAIN_ARRAY_LARGE` as discriminators. Please, fix also with the other collection types (sequences and maps).

These TypeIdentifiers should be checked as well.